backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/ModularContractService.java

```diff
@@ -399,4 +406,73 @@ public void cancelContractsOnOrderVoidIfNeededAndAllowed(@NonNull final OrderId
 													.build());
 		}
 	}
+
+	public void closeContractsOnOrderCloseIfNeededAndAllowed(@NonNull final I_C_Order orderRecord)
+	{
+		final List<de.metas.interfaces.I_C_OrderLine> orderLines = orderBL.retrieveOrderLines(orderRecord);
+		final Set<FlatrateTermId> modularContractIds = orderLines.stream()
+				.filter(this::isModularContractLine)
+				.map(I_C_OrderLine::getC_Order_ID)
+				.map(OrderId::ofRepoId)
+				.map(flatrateBL::getByOrderId)
```

Comment:
i hope this is not a frequent operation, else here we would have a beauty SQL N+1 issue....

Reply:
anyways, maybe would be nice to refactor this part and introduce some getModularContractIds(I_C_Order) method